transfer/incrementor.go

```diff
@@ -98,28 +105,75 @@ func NewGasPriceIncremenetor(cfg GasIncrementorConfig, storage Storage, cl Multi
 // It will query the given storage for any entries that it needs to check
 // for gas increase, trying to check their status.
 func (i *GasPriceIncremenetor) Run() {
-	process := func(txs []Transaction) {
-		for _, tx := range txs {
-			switch tx.State {
-			case TxStateFailed, TxStateSucceed:
-				// Force skip transactions that are finalized.
-			default:
-				i.tryWatch(tx)
-			}
-		}
-	}
-
 	for {
 		select {
 		case <-i.stop:
 			return
 
 		case <-time.After(i.cfg.PullInterval):
-			txs, err := i.storage.GetIncrementorTransactionsToCheck(i.signers.getSigners())
+			txs, err := i.storage.GetIncrementorTransactionsToCheck(i.cfg.PullEntryCount, i.signers.getSigners())
 			if err != nil {
 				continue
 			}
-			process(txs)
+
+			i.queueWork(txs)
+		}
+	}
+}
+
+func (i *GasPriceIncremenetor) watchOrIncrease(work chan Transaction) {
+	for tx := range work {
```

Comment:
This needs to have an exit strategy. If you queue 1000 tx, and call stop, we will still be processigng them.

Reply:
This is actually alright because the max queue size can't be more than 1. Workers only have 1 item in queue and they pick it up as they become free then next item is queued. So even if `stop` is closed, we shouldnt be affected.